### PR TITLE
Adds config option to prevent repeating maps

### PR DESCRIPTION
**Pull request by S34NW**
## What Does This PR Do
Adds a config option that, when enabled, prevents the current map from showing up in the map vote options

## Why It's Good For The Game
More flexibility to allow different map rotation styles

## Testing
All i hear is the map vote noise now

## Changelog
:cl:
add: Added a config option to prevent the same map from being voted on twice in a row
/:cl:

### CODE REVIEW

**Comment by YoureNotTheSharpest** [comment]
Do it

**Comment by Scribble-Sheep** [comment]
Do it, u wont

**Comment by russetdapone** [comment]
merge

File: code/controllers/configuration/sections/vote_configuration.dm [hunk 1]
```diff
@@ -12,12 +12,15 @@
 	var/disable_default_vote = TRUE
 	/// Enable map voting?
 	var/enable_map_voting = FALSE
+	/// If TRUE, you will not be able to vote for the current map
+	var/non_repeating_maps = FALSE
```
Comment (AffectedArc07) [inline code/controllers/configuration/sections/vote_configuration.dm:16]: Default this to true.

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

File: config/example/config.toml [hunk 2]
```diff
@@ -814,7 +814,8 @@ prevent_dead_voting = false
 disable_default_vote = true
 # Enable voting for a map on end round
 enable_map_voting = false
-
+# Set to true to prevent the same map from being voted on twice in a row
+non_repeating_maps = false
```
Comment (S34NW) [inline config/example/config.toml:818]:
```suggestion
non_repeating_maps = true
```

**Comment by Samman166** [comment]
Forever box is dead 🦀

**Comment by Hematoma** [comment]
I want to play but after three box rounds in a row it just hurts.

**Review by hal9000PR — APPROVED** [review]

**Review by AffectedArc07 — APPROVED** [review]
Seems fine but I reviewed this on a phone on a train so good luck.

**Comment by Scribble-Sheep** [comment]
Omg he actually did it

**Pull request merged**